pyomo/contrib/pynumero/asl.py

```diff
@@ -364,7 +365,8 @@ def eval_f(self, x):
         res = self.ASLib.EXTERNAL_AmplInterface_eval_f(
             self._obj, x, self._nx, ctypes.byref(sol)
         )
-        assert res, "Error in AMPL evaluation"
+        if not res:
+            raise PyNumeroEvaluationError("Error in AMPL evaluation")
```

Comment:
Likely not your responsibility, but is there a way to make this error more explicit? It seems vague in its current state.

Reply:
We could say e.g. "AMPL evaluation error in objective function". Or were you thinking of more explicit in some other way?

Reply:
I was thinking, is there a way to make the error more helpful to the user? Like, is this an error that they can fix (e.g., is it an error in their model)? If so, are there any hints we can give them about _how_ to fix it?

Reply:
Nope! We have no idea what happened inside ASL. Even if we could narrow down exactly what constraint (or objective) the error occurred in (which I have a vague proposal for in #2899), we would have no idea what expression caused ASL to fail. The best we could do is try to evaluate the Pyomo constraints (which we don't have access to at this layer of the interface) and hope that Pyomo raises an error about the same expression that ASL didn't like.

Reply:
_Lovely_... in which case, IGNORE this comment.